DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/part1_pointers/src/mmhost.cpp

```diff
@@ -0,0 +1,97 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+using namespace sycl;
+
+struct PointerIP{
+  
+  //Declare the pointer interfaces to be used in this kernel,
+  //look at the other kernals to compare the difference 
+  int *x; 
+  int *y; 
+  int *z;
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      z[i] = x[i] + y[i];
+    }
+  }
+};
+
+int main(void){
+#if FPGA_SIMULATOR
+  auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+  auto selector = sycl::ext::intel::fpga_selector_v;
+#else  // #if FPGA_EMULATOR
+  auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+  bool passed = true;
+
+  try {
+    // create the device queue
+    sycl::queue q(selector, fpga_tools::exception_handler,
+                  sycl::property::queue::enable_profiling{});
+
+    // Print out the device information.
+    sycl::device device = q.get_device();
+    std::cout << "Running on device: "
+              << q.get_device().get_info<sycl::info::device::name>().c_str()
+              << std::endl;
+
+    // make sure the device supports USM host allocations
+    if (!device.has(sycl::aspect::usm_host_allocations)) {
+      std::terminate();
+    }
```

Comment:
Why is this needed for a IPA tutorial?

Reply:
Is this not needed for the malloc_shared calls?

Reply:
Only for Full system flow - to check that the targeted board supports host allocations. For IPA we define the device specs and we support host and shared USM (but not device USM so malloc_device) will not work.

Reply:
Its not, removing this from all related files